src/js/aos.js

```diff
@@ -119,9 +127,10 @@ const init = function init(settings) {
    * Set global settings on body, based on options
    * so CSS can use it
    */
-  document.querySelector('body').setAttribute('data-aos-easing', options.easing);
-  document.querySelector('body').setAttribute('data-aos-duration', options.duration);
-  document.querySelector('body').setAttribute('data-aos-delay', options.delay);
+  var tagConfig = options.root != document ? options.root.querySelector('div'): document.querySelector('body');
```

Comment:
I can understand why we target `body` when the root is window (which is not an HTMLElement). But why searching for inner `div` for others cases, instead of using `root`. We should _at least_ checking if `root` is an `HTMLElement` and set tags on it, if not simply expecting it to be one (with an exception for `window` and `document`).